### ISSUE TITLE
Use npm to pull in reveal?

### ISSUE THREAD

**Issue by britter** [codecentric/codecentric-reveal-template#2]
Hi,

would it be possible/feasable to pull in reveal.js using npm instead of having a copy in this repo? This way it would be easier to upgrade the reveal version.

**Comment by dzenzes** [comment]
Something like [this](https://github.com/dmies/codecentric-reveal-template)?

**Comment by rfalke** [comment]
The idea is that the slides work without any external dependencies. So that you can present the slides even if you don't have network access at the customer. I lot of things can happen when you present slides and the fact that the slides are self-contained is just one less problem to worry about. So for my use case the answer is a no.

What we can do is to use two branches: 'selfcontained' and 'networkinstalled'.

**Comment by britter** [comment]
@rfalke you can simply do a `npm install` at home before you go to the customer.

**Comment by bripkens** [comment]
You could also manage the dependencies via NPM and check the `node_modules` into the repository. Both problems solved :).

**Comment by dzenzes** [comment]
I wouldn't checkin the `node_modules` folder. The user can just install everything via npm and it works.

**Comment by britter** [comment]
I agree with @dmies, but maybe we really need to separate templates (which would make the NPM version the third version we have :))

**Comment by bripkens** [comment]
Actually, there is no technical reason not to check in the `node_modules/` dir in this case. Yes, it doesn't feel good and it might not be up to our usual standards, but it would reduce the barrier to entry and ensure that everyone is using a reveal.js version which is compatible to the template.

Also, non-Node.js lovers wouldn't be required to install Node.js just so that they can install NPM just so that they can install reveal.js just so that they can work on / present some slides.

**Comment by rfalke** [comment]
Feel free to do what you want. You have apparently have a stronger opinion than I do.

I can always create a branch from some older version.

**Comment by britter** [comment]
> Actually, there is no technical reason not to check in the node_modules/ dir in this case.

I think it will lead to enormous diffs when upgrading reveal.js version.

**Comment by dzenzes** [comment]
If you add a dependency that needs to be compiled (like sass) you have to use npm anyway.
